### PR TITLE
Help to not merge during freeze

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

File: .github/workflows/merge-freeze.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+jobs:
+  block-merge-during-freeze:
+    name: Block merge during freezes
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Run check
+        run: cat version.php | grep 'OC_VersionString' | grep -i -v 'RC'
```
Comment (szaimen) [inline .github/workflows/merge-freeze.yml:13]: It should probably check this file? https://raw.githubusercontent.com/nextcloud/server/master/version.php

Comment (nickvergessen) [reply .github/workflows/merge-freeze.yml:13]: No, it should be the local file, but i didn't check the path yet.

Comment (szaimen) [reply .github/workflows/merge-freeze.yml:13]: But what if the PR is old and the branch outdated?

Comment (nickvergessen) [reply .github/workflows/merge-freeze.yml:13]: Right, so curl it is... need to find the base branch option

File: .github/workflows/merge-freeze.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Pull request checks
+
+on: pull_request
```
Comment (szaimen) [inline .github/workflows/merge-freeze.yml:3]: I guess it should also be `on: push` (and maybe exclude master and stable branches...)

Comment (szaimen) [reply .github/workflows/merge-freeze.yml:3]: something like this:

```
    branches:    
    - '**'
    - '!master'
    - '!stable[12][0-9]'
```

Comment (nickvergessen) [reply .github/workflows/merge-freeze.yml:3]: pull_request means it runs on all pull requests, it is also what's used for the fixup check, so should be good enough

Comment (szaimen) [reply .github/workflows/merge-freeze.yml:3]: Okay

File: .github/workflows/merge-freeze.yml (hunk 1 above)
Comment (MichaIng) [reply .github/workflows/merge-freeze.yml:21]: Btw, `grep` takes files as arguments, so using `cat ... | grep ...` is generally unneeded overhead 😉:
```sh
grep 'OC_VersionString' version.php | grep -i -v 'RC'
```
or
```sh
! grep -i 'OC_VersionString.*RC' version.php
```

**Review by szaimen — APPROVED** [review]

**Pull request closed without merging**